datasets/large_spanish_corpus/large_spanish_corpus.py

```diff
@@ -0,0 +1,124 @@
+
+import datasets
+
+
+_CITATION = """\
+@dataset{jose_canete_2019_3247731,
+  author       = {José Cañete},
+  title        = {Compilation of Large Spanish Unannotated Corpora},
+  month        = may,
+  year         = 2019,
+  publisher    = {Zenodo},
+  doi          = {10.5281/zenodo.3247731},
+  url          = {https://doi.org/10.5281/zenodo.3247731}
+}
+"""
+
+_DESCRIPTION = """\
+The Large Spanish Corpus is a compilation of 15 unlabelled Spanish corpora spanning Wikipedia to European parliament \
+notes. Each config contains the data corresponding to a different corpus. For example, "all_wiki" only includes \
+examples from Spanish Wikipedia. By default, the config is set to "combined" which loads all the corpora; with this \
+setting you can also specify the number of samples to return per corpus by configuring the "split" argument.
+"""
+
+_HOMEPAGE = "https://github.com/josecannete/spanish-corpora"
+
+_LICENSE = ""
+
+_URL = "https://zenodo.org/record/3247731/files/raw.tar.bz2"
+
+_CORPORA = [
+    "JRC",
+    "EMEA",
+    "GlobalVoices",
+    "ECB",
+    "DOGC",
+    "all_wikis",
+    "TED",
+    "multiUN",
+    "Europarl",
+    "NewsCommentary11",
+    "UN",
+    "EUBookShop",
+    "ParaCrawl",
+    "OpenSubtitles2018",
+    "DGT",
+]
+
+_CORPORA_FILEPATHS = {corpus: os.path.join("spanish-corpora/raw", f"{corpus}.txt") for corpus in _CORPORA}
+
+_VERSION = "1.1.0"
+
+_COMBINED = "combined"
+
+
+class LargeSpanishCorpusConfig(datasets.BuilderConfig):
+    def __init__(self, corpora=None, **kwargs):
+        super(LargeSpanishCorpusConfig, self).__init__(version=datasets.Version(_VERSION, ""), **kwargs)
+        self.corpora = corpora
+
+    @property
+    def filepaths(self):
+        return [_CORPORA_FILEPATHS[corpus] for corpus in self.corpora]
+
+
+class LargeSpanishCorpus(datasets.GeneratorBasedBuilder):
+    """The Large Spanish Corpus."""
+
+    BUILDER_CONFIGS = [
+        LargeSpanishCorpusConfig(name=corpus, corpora=[corpus], description=f"Spanish examples in corpus {corpus}.")
+        for corpus in _CORPORA
+    ] + [
+        LargeSpanishCorpusConfig(
+            name=_COMBINED, corpora=_CORPORA, description=f"Complete Spanish dataset with all corpora."
+        )
+    ]
+
+    DEFAULT_CONFIG_NAME = _COMBINED
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(name=corpus, gen_kwargs={"filepath": os.path.join(data_dir, filepath)})
+            for corpus, filepath in zip(self.config.corpora, self.config.filepaths)
+        ]
```

Comment:
Splits are supposed to be used for train-dev-test splits for example depending of the model evaluations you want to make.
Here if I understand correctly the different corpora are just defined by their sources.
For different sources we usually use dataset configurations.

So I would keep all the configurations you defined, but I would change the splits.
Since the data is not split, please use one split `name=datasets.Split.TRAIN` for all of them.
For the combined config, the different corpora can simply be concatenated.

Does that make sense ?

Reply:
Thanks for the feedback!

Sorry for abusing the splits - I'll fix it as you suggest. When you say "the different corpora can simply be concatenated" are you saying I should change the existing config for combined? If yes, would you mind elaborating on what you'd like to see changed?

Reply:
OK I've updated the logic along the lines you suggested and am currently generating the `dataset_infos.json` and dummy data (takes a looong time with this corpus 😅). So the tests will fail until then ...

Is this what you had in mind?

Reply:
Yes that's exactly what I had in mind, thank you !

Reply:
Cool! Will let you know when the PR is ready for a final pass :)

Reply:
Looks like the tests are passing with the new changes / files - ready for final review!